lib/heap_sort.rb

```diff
@@ -1,8 +1,24 @@
-
+require 'pry'
+require_relative 'min_heap'
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  O(n) followed by O(log n) beacause going through entire array for first operation
+# Space Complexity: I am attempting to do it in O(1)
```

Comment:
Since you're creating a heap and building it up.  This is O(n) space complexity.